### PR TITLE
resolved issue in agent where crosstalk is possible

### LINKED ISSUE #2163: When agents failover crosstalk between environments is possible
The following was observed

```
[centos@iso3-test ~]$ tail -f /var/log/inmanta/agent-282d77e8-906e-4f77-95e5-7e377acaf20f.log
2020-06-05 12:15:43,661 INFO     inmanta.agent.handler Sending std::AgentConfig[internal,agentname=localhost]: success
2020-06-05 12:15:45,959 INFO     inmanta.agent.agent Agent internal got a trigger to update in environment 282d77e8-906e-4f77-95e5-7e377acaf20f
2020-06-05 12:15:48,496 INFO     inmanta.agent.agent Agent internal got a trigger to update in environment 282d77e8-906e-4f77-95e5-7e377acaf20f
2020-06-05 12:17:08,384 INFO     inmanta.agent.agent Agent internal got a trigger to update in environment 719c7ad5-6657-444b-b536-a27174cb7498
2020-06-05 12:20:00,342 INFO     inmanta    Agent Shutdown complete
```

The agent received a call for the wrong environment. The agent refuses to execute this, so this is 'safe' but it stops the orchestrator from making proper progress.

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Fixes the agent crosstalk problem

closes #2163

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: tests/test_agent_manager.py [hunk 1]
```diff
@@ -336,10 +369,27 @@ async def dummy_status():
...
+                "last_failover": UNKWN,
+                "primary": UNKWN,
+                "environment": env4.id,
+                "state": "up",
+            },
+            {
+                "name": "agent3",
+                "paused": False,
+                "last_failover": UNKWN,
+                "primary": UNKWN,
```
Comment (arnaudsjs) [inline tests/test_agent_manager.py:385]: `last_failover` and `primary` should be equal to an empty string right? `agent3` on environment `env4` never had a session.

Comment (wouterdb) [reply tests/test_agent_manager.py:385]: thx

Comment (arnaudsjs) [reply tests/test_agent_manager.py:385]: This one was not updated?

Comment (wouterdb) [reply tests/test_agent_manager.py]: push had failed, now it should be good

**Pull request merged**